### ISSUE TITLE
Chunk list when starting a live stream

### ISSUE THREAD

**Issue by timpalander** [mangui/flashls#198]
Hi,

We have a custom player using OSMF and FlashHLS. We're trying to track down an issue that occurs when we start the player before starting a live stream. In this scenario we get an I/O error on the playlist.m3u8 as expected (since it doesn't exist in Wowza until the stream is actually running), i then use a timer to retry getting the playlist on intervals until it responds. When the playlist exists the stream starts playing. In this scenario the buffer length will only go to about 8-12 and keep emptying to 0.1-2, all while buffering true/false and OSMF Unpause keeps spamming the log. This goes on for a while and eventually the buffer length reaches 0 and i get a stop event. After the stop event i reload the playlist and play again, and then everything runs just fine. Buffer length is up to 15-30 and way less Unpause and Buffer events spamming.

Our best bet so far is that we get stuck in some state when loading an early playlist that's partially incomplete. It seems like you have to wait 30-50 seconds in to the live stream before even trying to play that playlist, and then everything runs just fine.

-----------------------------------------------------------
(Bad) Here's the log for the first chunklist in the scenario where we start the player before the stream:
-----------------------------------------------------------
DEBUG:adaptive playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=250000
chunklist_w2110796494.m3u8?id=&s_referrer=&s_type=

DEBUG:level 0 playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:NO
#EXT-X-TARGETDURATION:11
#EXT-X-MEDIA-SEQUENCE:2
#EXTINF:10.148,
media_w2110796494_2.ts?id=&s_referrer=&s_type=

DEBUG:updateFragments: unknown PTS info for this level
DEBUG:Level 0 Live Playlist parsing finished: reload in 5028 ms
DEBUG:first level filled with at least 1 fragment, notify event

-----------------------------------------------------------
(Good) Here's the log for when i start the player after the stream has been running for about 50 seconds:
-----------------------------------------------------------
DEBUG:adaptive playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=933251,CODECS="avc1.77.41, mp4a.40.34",RESOLUTION=960x720
chunklist_w428478978.m3u8?id=&s_referrer=&s_type=

DEBUG:level 0 playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:NO
#EXT-X-TARGETDURATION:11
#EXT-X-MEDIA-SEQUENCE:6
#EXTINF:10.097,
media_w428478978_6.ts?id=&s_referrer=&s_type=
#EXTINF:10.047,
media_w428478978_7.ts?id=&s_referrer=&s_type=
#EXTINF:7.585,
media_w428478978_8.ts?id=&s_referrer=&s_type=

DEBUG:updateFragments: unknown PTS info for this level
DEBUG:Level 0 Live Playlist parsing finished: reload in 9178 ms
DEBUG:first level filled with at least 1 fragment, notify event

**Comment by mangui** [comment]
Hi @timpalander have you stripped the logs ? I would need to see the load*fragment() message in the debug logs to understand what is going on. (this should happen in your good scenario)
you can also eventually enable debug2 logs as well, to avoid loosing any meaningful info ...

**Comment by timpalander** [comment]
-------------
Bad
-------------
DEBUG:adaptive playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=250000
chunklist_w2110796494.m3u8?id=&s_referrer=&s_type=

DEBUG:level 0 playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:NO
#EXT-X-TARGETDURATION:11
#EXT-X-MEDIA-SEQUENCE:2
#EXTINF:10.148,
media_w2110796494_2.ts?id=&s_referrer=&s_type=

DEBUG:updateFragments: unknown PTS info for this level
DEBUG:Level 0 Live Playlist parsing finished: reload in 5028 ms
DEBUG:first level filled with at least 1 fragment, notify event
DEBUG:HLSNetStreamLoadTrait()
DEBUG:HLSMediaElement:processReadyState
DEBUG:HLSBufferTrait()
onBufferTimeChange: bufferTrait.bufferTime: 8
DEBUG:HLSTimeTrait()
DEBUG:HLSDisplayObjectTrait()
DEBUG:HLSPlayTrait()
INFO:HLSPlayTrait:playStateChangeStart:playing
INFO:HLSNetStream:play(-1)
INFO:HLSNetStream:seek(-1)
DEBUG:[SEEK_STATE] from IDLE to SEEKING
DEBUG:[PLAYBACK_STATE] from IDLE to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
[SLog]:[INFO]: Playstate changed to: playing
[SLog]:[INFO]: Metrics update interval set to: 180 seconds.
DEBUG:HLSSeekTrait()
DEBUG:HLSAlternativeAudioTrait()
Thu Feb 5 2015 03:00:13 PM.423 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: BEGIN
Thu Feb 5 2015 03:00:13 PM.423 [DEBUG] [org.osmf.layout.LayoutRendererBase] VALIDATING LAYOUT
Thu Feb 5 2015 03:00:13 PM.425 [INFO] [org.osmf.media.videoClasses.VideoSurface] switchRenderer. currentVideoRenderer = null; the new renderer = [object StageVideo]
Thu Feb 5 2015 03:00:13 PM.427 [DEBUG] [org.osmf.layout.LayoutRendererBase] LAYOUT VALIDATED
Thu Feb 5 2015 03:00:13 PM.427 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: END
DEBUG:loadfirstfragment(-1)
DEBUG:loadfirstfragment : requested position:-1,seek position:0
DEBUG:Loading       2 of [2,2],level 0
DEBUG:loading fragment:http://80.75.24.183/edge/_dhz52oqh_/dhz52oqh/media_w2110796494_2.ts?id=&s_referrer=&s_type=
DEBUG:probe fragment type
DEBUG:AAC/MP3/TS match:false/false/true
DEBUG:TS: PAT found.PMT PID:4095
DEBUG:TS: PMT found
DEBUG:TS: Selected ID3 PID: 258
DEBUG:TS: Selected video PID: 256
DEBUG:TS: Found 1 audio tracks
DEBUG:HLSDynamicStreamTrait:_audioTrackListChangedHandler
DEBUG:HLSDynamicStreamTrait:_audioTrackChangedHandler
INFO:Setting audio track to 0
DEBUG:TS: selected MP3 PID: 257
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: width/height:960/720
DEBUG:AVC: H264 Main level 41
DEBUG:1 tags extracted
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:53 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:0.00/1.56/0.00/true
DEBUG:AutoBufferManager:minBufferLength:1.1246006389776357
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: H264 Main level 41
DEBUG:29 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:1.56/0.86/0.00/false
DEBUG:AutoBufferManager:minBufferLength:1.0120513049314788
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:33 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:2.43/1.02/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.9179211847169663
Thu Feb 5 2015 03:00:13 PM.811 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:13 PM.811 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Seek.Notify
DEBUG:[SEEK_STATE] from SEEKING to SEEKED
DEBUG:playing fragment(level/sn/cc):0/2/0
DEBUG:custom tag:#EXT-X-ALLOW-CACHE:NO
DEBUG:custom tag:#EXTINF:10.148,
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:22 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:3.45/0.70/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.8622225822380442
Thu Feb 5 2015 03:00:13 PM.845 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Buffer.Full
Thu Feb 5 2015 03:00:13 PM.874 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Video.DimensionChange
INFO:HLSDisplayObjectTrait:setMediaSize(960,720)
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: H264 Main level 41
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:78 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:4.15/2.35/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.8212171087580632
Thu Feb 5 2015 03:00:13 PM.900 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: BEGIN
Thu Feb 5 2015 03:00:13 PM.900 [DEBUG] [org.osmf.layout.LayoutRendererBase] VALIDATING LAYOUT
Thu Feb 5 2015 03:00:13 PM.901 [DEBUG] [org.osmf.layout.LayoutRendererBase] LAYOUT VALIDATED
Thu Feb 5 2015 03:00:13 PM.901 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: END
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: H264 Main level 41
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:85 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:6.50/2.67/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.623762544965971
DEBUG:resume playback
Thu Feb 5 2015 03:00:13 PM.931 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:loading completed
DEBUG:Loading       duration/RTT/length/speed:375/68/1072916/22352 kb/s
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:26 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:9.17/0.78/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.7152025827223647
DEBUG:TS: flushing demux
DEBUG:TS: complete Audio PES found at end of segment, parse it
DEBUG:TS: complete AVC PES found at end of segment, parse it
DEBUG:AVC: AUD, NDR slices
DEBUG:TS: flush 3 tags
DEBUG:3 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:9.95/0.08/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.7115741277807177
DEBUG:TS: parsing complete
DEBUG:m/M audio PTS:5228206/5238236
DEBUG:m/M video PTS:5228268/5238365
DEBUG:Delta audio/video m/M PTS:62/129
DEBUG:Total Process duration/length/bw:491/1072916/17072 kb/s
DEBUG:Loaded        2 of [2,2],level 0 m/M PTS:5228206/5238236
DEBUG:AutoBufferManager:minBufferLength:0.7135022481168437
DEBUG:loadnextfragment()
DEBUG:level 0 playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:NO
#EXT-X-TARGETDURATION:11
#EXT-X-MEDIA-SEQUENCE:2
#EXTINF:10.148,
media_w2110796494_2.ts?id=&s_referrer=&s_type=

DEBUG:updateFragments: found PTS info from previous playlist,seqnum/PTS:2/5228206
DEBUG:Level 0 Live Playlist parsing finished: reload in 10083 ms
DEBUG:loadnextfragment()
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:21 PM.307 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:21 PM.413 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:21 PM.542 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:21 PM.663 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:21 PM.792 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:21 PM.913 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.042 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.160 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.297 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.410 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.546 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.663 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.796 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:22 PM.915 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:23 PM.048 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:23 PM.162 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:23 PM.301 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
DEBUG:resume playback
Thu Feb 5 2015 03:00:23 PM.413 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:[PLAYBACK_STATE] from PLAYING to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
Thu Feb 5 2015 03:00:24 PM.162 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:24 PM.245 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Buffer.Empty
Thu Feb 5 2015 03:00:24 PM.301 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:24 PM.411 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:24 PM.546 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:24 PM.662 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:24 PM.796 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:24 PM.916 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.047 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.163 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.297 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.412 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.547 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.663 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.797 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:25 PM.912 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.047 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.160 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.297 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.416 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.546 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.663 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.796 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:26 PM.911 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.046 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.163 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.296 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.413 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.546 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.662 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.795 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:27 PM.914 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:28 PM.046 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:28 PM.163 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:28 PM.295 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:28 PM.414 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:00:28 PM.547 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
DEBUG:level 0 playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:NO
#EXT-X-TARGETDURATION:11
#EXT-X-MEDIA-SEQUENCE:2
#EXTINF:10.148,
media_w2110796494_2.ts?id=&s_referrer=&s_type=
#EXTINF:10.046,
media_w2110796494_3.ts?id=&s_referrer=&s_type=

DEBUG:updateFragments: found PTS info from previous playlist,seqnum/PTS:2/5228206
DEBUG:Level 0 Live Playlist parsing finished: reload in 10038 ms
DEBUG:loadnextfragment()
DEBUG:Loading       3 of [2,3],level 0
DEBUG:loading fragment:http://80.75.24.183/edge/_dhz52oqh_/dhz52oqh/media_w2110796494_3.ts?id=&s_referrer=&s_type=
Thu Feb 5 2015 03:00:28 PM.663 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: width/height:960/720
DEBUG:AVC: H264 Main level 41
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
-------------
Good
-------------
DEBUG:adaptive playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=933251,CODECS="avc1.77.41, mp4a.40.34",RESOLUTION=960x720
chunklist_w428478978.m3u8?id=&s_referrer=&s_type=

DEBUG:level 0 playlist:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:NO
#EXT-X-TARGETDURATION:11
#EXT-X-MEDIA-SEQUENCE:6
#EXTINF:10.097,
media_w428478978_6.ts?id=&s_referrer=&s_type=
#EXTINF:10.047,
media_w428478978_7.ts?id=&s_referrer=&s_type=
#EXTINF:7.585,
media_w428478978_8.ts?id=&s_referrer=&s_type=

DEBUG:updateFragments: unknown PTS info for this level
DEBUG:Level 0 Live Playlist parsing finished: reload in 9178 ms
DEBUG:first level filled with at least 1 fragment, notify event
DEBUG:HLSNetStreamLoadTrait()
DEBUG:HLSMediaElement:processReadyState
DEBUG:HLSBufferTrait()
onBufferTimeChange: bufferTrait.bufferTime: 8
DEBUG:HLSTimeTrait()
DEBUG:HLSDisplayObjectTrait()
DEBUG:HLSPlayTrait()
INFO:HLSPlayTrait:playStateChangeStart:playing
INFO:HLSNetStream:play(-1)
INFO:HLSNetStream:seek(-1)
DEBUG:[SEEK_STATE] from IDLE to SEEKING
DEBUG:[PLAYBACK_STATE] from IDLE to PLAYING_BUFFERING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(true)
[SLog]:[INFO]: Playstate changed to: playing
[SLog]:[INFO]: Metrics update interval set to: 180 seconds.
DEBUG:HLSSeekTrait()
DEBUG:HLSAlternativeAudioTrait()
Thu Feb 5 2015 03:01:17 PM.270 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: BEGIN
Thu Feb 5 2015 03:01:17 PM.270 [DEBUG] [org.osmf.layout.LayoutRendererBase] VALIDATING LAYOUT
Thu Feb 5 2015 03:01:17 PM.272 [INFO] [org.osmf.media.videoClasses.VideoSurface] switchRenderer. currentVideoRenderer = null; the new renderer = [object StageVideo]
Thu Feb 5 2015 03:01:17 PM.275 [DEBUG] [org.osmf.layout.LayoutRendererBase] LAYOUT VALIDATED
Thu Feb 5 2015 03:01:17 PM.275 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: END
DEBUG:loadfirstfragment(-1)
DEBUG:loadfirstfragment : requested position:-1,seek position:0
DEBUG:Loading       6 of [6,8],level 0
DEBUG:loading fragment:http://80.75.24.183/edge/_dhz52oqh_/dhz52oqh/media_w428478978_6.ts?id=&s_referrer=&s_type=
DEBUG:probe fragment type
DEBUG:AAC/MP3/TS match:false/false/true
DEBUG:TS match + H264 signaled in Manifest, use TS demuxer
DEBUG:TS: PAT found.PMT PID:4095
DEBUG:TS: PMT found
DEBUG:TS: Selected ID3 PID: 258
DEBUG:TS: Selected video PID: 256
DEBUG:TS: Found 1 audio tracks
DEBUG:HLSDynamicStreamTrait:_audioTrackListChangedHandler
DEBUG:HLSDynamicStreamTrait:_audioTrackChangedHandler
INFO:Setting audio track to 0
DEBUG:TS: selected MP3 PID: 257
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: width/height:960/720
DEBUG:AVC: H264 Main level 41
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:3 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:0.00/0.00/0.00/true
DEBUG:AutoBufferManager:minBufferLength:60
Thu Feb 5 2015 03:01:17 PM.650 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Pause.Notify
Thu Feb 5 2015 03:01:17 PM.651 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Seek.Notify
DEBUG:[SEEK_STATE] from SEEKING to SEEKED
DEBUG:playing fragment(level/sn/cc):0/6/0
DEBUG:custom tag:#EXT-X-ALLOW-CACHE:NO
DEBUG:custom tag:#EXTINF:10.097,
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: H264 Main level 41
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:87 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:0.00/2.66/0.00/false
DEBUG:AutoBufferManager:minBufferLength:1.867103154692964
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: H264 Main level 41
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:83 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:2.66/2.59/0.00/false
DEBUG:AutoBufferManager:minBufferLength:1.0320974635154396
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:56 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:5.25/1.73/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.8545122489547939
DEBUG:loading completed
DEBUG:Loading       duration/RTT/length/speed:427/273/1058440/19365 kb/s
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: H264 Main level 41
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:98 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:6.98/2.98/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.7240895382378612
DEBUG:TS: flushing demux
DEBUG:TS: complete Audio PES found at end of segment, parse it
DEBUG:TS: complete AVC PES found at end of segment, parse it
DEBUG:AVC: AUD, NDR slices
DEBUG:TS: flush 3 tags
DEBUG:3 tags extracted
DEBUG:Loaded position/duration/sliding/discontinuity:9.95/0.08/0.00/false
DEBUG:AutoBufferManager:minBufferLength:0.7196161711656031
DEBUG:TS: parsing complete
DEBUG:m/M audio PTS:5268485/5278517
DEBUG:m/M video PTS:5268555/5278601
DEBUG:Delta audio/video m/M PTS:70/84
DEBUG:Total Process duration/length/bw:454/1058440/18214 kb/s
DEBUG:Loaded        6 of [6,8],level 0 m/M PTS:5268485/5278517
DEBUG:AutoBufferManager:minBufferLength:0.7219250119439493
DEBUG:loadnextfragment()
DEBUG:Loading       7 of [6,8],level 0
DEBUG:loading fragment:http://80.75.24.183/edge/_dhz52oqh_/dhz52oqh/media_w428478978_7.ts?id=&s_referrer=&s_type=
Thu Feb 5 2015 03:01:17 PM.821 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Video.DimensionChange
INFO:HLSDisplayObjectTrait:setMediaSize(960,720)
Thu Feb 5 2015 03:01:17 PM.826 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: BEGIN
Thu Feb 5 2015 03:01:17 PM.826 [DEBUG] [org.osmf.layout.LayoutRendererBase] VALIDATING LAYOUT
Thu Feb 5 2015 03:01:17 PM.827 [DEBUG] [org.osmf.layout.LayoutRendererBase] LAYOUT VALIDATED
Thu Feb 5 2015 03:01:17 PM.827 [DEBUG] [org.osmf.layout.LayoutRendererBase] ON EXIT FRAME: END
Thu Feb 5 2015 03:01:17 PM.859 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Buffer.Full
DEBUG:AVC: AUD, SPS, PPS, IDR slices
DEBUG:AVC: width/height:960/720
DEBUG:AVC: H264 Main level 41
DEBUG:1 tags extracted
DEBUG:resume playback
Thu Feb 5 2015 03:01:17 PM.898 [INFO] [org.osmf.net.NetStreamLoadTrait] NetStream.Unpause.Notify
DEBUG:[PLAYBACK_STATE] from PLAYING_BUFFERING to PLAYING
DEBUG:HLSBufferTrait:_stateChangedHandler:setBuffering(false)
DEBUG:AVC: no NALU slices found
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices
DEBUG:AVC: AUD, NDR slices

**Comment by timpalander** [comment]
When the initial chunklist only contains one chunk, it seems like it will never put more than one chunk in the buffer at once. Same with 2 or 3. When i eventually get the stop event and reload everything i get 3 chunks, and that's how much i get in the buffer as well. The last fragment before i get the stop event always seem to be around 7 in duration.

Here's a few picks from things i'm tracing:
frags from ManifstLoader::_parseLevelPlaylist
bufferLength() from HLSNetStream

Here's what happens
-----------------------

Fragment (seqnum: 1, start_time:0, duration:11.551)
bufferLength is: ~10, counts down to ~0.05

Fragment (seqnum: 1, start_time:0, duration:11.551),Fragment (seqnum: 2, start_time:11.551, duration:10.533)
bufferLength is: ~10, counts down to ~0.05

Fragment (seqnum: 1, start_time:0, duration:11.52),Fragment (seqnum: 2, start_time:11.52, duration:10.533),Fragment (seqnum: 3, start_time:22.052999999999997, duration:10.534)
bufferLength is: ~10, counts down to ~0.05

Fragment (seqnum: 2, start_time:0, duration:10.579),Fragment (seqnum: 3, start_time:10.579, duration:10.534),Fragment (seqnum: 4, start_time:21.113, duration:10.533)

INFO:HLSPlayTrait:playStateChangeStart:stopped
INFO:HLSNetStream:close
INFO:HLSSeekTrait:seekingChangeStart(newSeeking/time):(true/0)
INFO:HLSNetStream:seek(0)
(me) [SLog]:[INFO]: Retrying to play in: 1 seconds

Fragment (seqnum: 3, start_time:0, duration:10.534),Fragment (seqnum: 4, start_time:10.534, duration:10.533),Fragment (seqnum: 5, start_time:21.067, duration:7.022)
INFO:HLSPlayTrait:playStateChangeStart:playing
bufferLength is: ~30, counts down to ~20

It's now stable and keeps a healthy buffer.

**Comment by mangui** [comment]
do you mean that fragments are not getting loaded fast enough ? most probably it would be because the live playlist is not refreshed fast enough.
in that case , there might be an issue with small live playlist as currently the reload timer is set to TimeOut after average segment duration.
see
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/loader/LevelLoader.as#L165
this needs to be refined IMHO, to something like : 
(end of playlist - playhead position) divided by 2,
you can eventually try on your side to adjust it to
```
 var timeout : Number = Math.max(100, _reload_playlists_timer + 1000 * _levels[level].averageduration/ 2 - getTimer());
```
this will spam a little bit your server. but adjusting the timer logic to the distance to the end of the playlist should help anyway.

**Comment by mangui** [comment]
Hi @timpalander , have you tried adjusting the timeout ?
does it fix your issue ?

**Comment by mangui** [comment]
Hi @timpalander you might recheck with latest flashls/dev, I did some changes so that live playlist will be reloaded more quickly when playback position is near the edge of the live playlist, this should improve the behaviour in case of live quick start playback.

**Comment by timpalander** [comment]
Hi. Thank you! I will try it out and see if it makes a difference. I didn't experience any difference when i tried to change it myself as you suggested.

**Comment by mangui** [comment]
@timpalander plz tell me if you still observe your live quick start issue. if still observed, I would need a log file
tks
Mangui

**Comment by timpalander** [comment]
I'm getting another issue with the Dev branch now that prevents me from testing this. When i start the player before the stream i get the "Cannot load M3U8: Error #2032" as expected. I then wait 20 seconds and try to play again, only this time around i get a null object reference on line 87 in StreamBuffer.as, _fragmentLoader is null. I traced it a bit and _streamBuffer.dispose() is done (setting _fragmentLoader to null) in HLS.dispose(), followed by _hlsNetStream.dispose_(), which in turn ends up trying to call _streamBuffer.stop().

HSL.as:74 -> HSL.dispose() -> _streamBuffer.dispose() (nulling _fragmentLoader) -> _hlsNetStream.dispose_() (calling _streamBuffer.stop())

This never happened before. Am i forgetting to clean up somewhere before i try to load the m3u8 again?

**Comment by timpalander** [comment]
It might have improved slightly. It seem harder to get it to come to a complete stop where it has to reload now. The issue still remains tho. If the player is loaded before the first 3 fragments are present, the buffer never recovers. I made a video of the issue: 

https://paland.ststage.se/widgets/embed/wv2isg4?auto_play=true

Here's what's going on: I start the player before the m3u8 exists. I then start a stream. The player finds it and starts playing. I let it play for a good while and as you can see the buffer length is very low. I click "Reload XML" which effectively re-initializes everything. After that the buffer is fine. If i restart the player instead of "Reload XML" it would have the same effect.

**Comment by mangui** [comment]
if you want to avoid buffer stalling, then I would recommend configuring
hls_minbufferlength to be set at around 2 fragments duration, so that even if you start playback with an empty/1 frag live playlist, the player will wait for 2 fragments to be buffered to start playback.

**Comment by mangui** [comment]
the other option would be to change this line
https://github.com/mangui/flashls/blob/915b8bb7435e93e8b9d658315e5854a3a364ee4d/src/org/mangui/hls/loader/LevelLoader.as#L179
and add a new settings parameter for live playlist 


```
 _canStart = (_levels[level].fragments.length >= HLSSettings.minLiveFragment);
if (_canStart) {
CONFIG::LOGGING {
Log.debug("first level filled with at least" + HLSSettings.minLiveFragment + "fragment(s), notify event");
}
_hls.dispatchEvent(new HLSEvent(HLSEvent.MANIFEST_LOADED, _levels));
}
```

**Comment by timpalander** [comment]
Increasing the minbufferlength seem to work pretty well, we'll do some tests with that and see how it works.

Trying the minLiveFragment option gave me this error:

TypeError: Error #1009: Cannot access a property or method of a null object reference.
	at org.mangui.hls.controller::AudioTrackController/_updateAudioTrackforLevel()[/Users/tim/Development/Solidtango/solidplayer/flashls/src/org/mangui/hls/controller/AudioTrackController.as:94]
	at org.mangui.hls.controller::AudioTrackController/_levelLoadedHandler()[/Users/tim/Development/Solidtango/solidplayer/flashls/src/org/mangui/hls/controller/AudioTrackController.as:71]
	at flash.events::EventDispatcher/dispatchEventFunction()
	at flash.events::EventDispatcher/dispatchEvent()
	at org.mangui.hls::HLS/dispatchEvent()[/Users/tim/Development/Solidtango/solidplayer/flashls/src/org/mangui/hls/HLS.as:67]
	at org.mangui.hls.loader::LevelLoader/_parseLevelPlaylist()[/Users/tim/Development/Solidtango/solidplayer/flashls/src/org/mangui/hls/loader/LevelLoader.as:188]
	at org.mangui.hls.playlist::Manifest/onLoadedData()[/Users/tim/Development/Solidtango/solidplayer/flashls/src/org/mangui/hls/playlist/Manifest.as:115]
	at org.mangui.hls.playlist::Manifest/_loaderHandler()[/Users/tim/Development/Solidtango/solidplayer/flashls/src/org/mangui/hls/playlist/Manifest.as:111]
	at flash.events::EventDispatcher/dispatchEventFunction()
	at flash.events::EventDispatcher/dispatchEvent()
	at flash.net::URLLoader/onComplete()

**Comment by mangui** [comment]
yes obviously for the minLiveFragment stuff you need to do other changes in HLSSettings also

**Comment by timpalander** [comment]
I actually did this to test, without changing HLSSettings:  _canStart = (_levels[level].fragments.length >= 2);

**Comment by mangui** [comment]
Hi @timpalander , I rechecked your stack dump and I dont understand how you could get this error by just modifying this line https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/loader/LevelLoader.as#L179

the issue happens here :
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/controller/AudioTrackController.as#L94

 if (_audioTracksfromManifest.length != audioTrackList.length) {

audioTrackList is initialized at the beginning of the same function, and
 _audioTracksfromManifest is initialized here :
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/controller/AudioTrackController.as#L63
this is triggered by MANIFEST_LOADED event which should be called prior to LEVEL_LOADED as you can see here:
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/loader/LevelLoader.as#L184-L187


and audioTrackList

**Comment by gkindel** [comment]
May be related, but I had to patch HLSTimeTrait to swallow the complete event that automatically fires in OSMF players when buffer reaches duration (when in LIVE).

**Comment by mangui** [comment]
plz reopen if issue is still repro/pending
